packages/@uppy/aws-s3-multipart/src/index.js

```diff
@@ -185,11 +180,17 @@ class HTTPCommunicationQueue {
     throwIfAborted(signal)
     const { uploadId, key } = await this.getUploadId(file, signal)
     throwIfAborted(signal)
-    const signature = await this.#fetchSignature(file, { uploadId, key, partNumber, body, signal })
-    throwIfAborted(signal)
-    return {
-      PartNumber: partNumber,
-      ...await this.#uploadPartBytes(signature, body, signal),
+    for (;;) {
+      const signature = await this.#fetchSignature(file, { uploadId, key, partNumber, body, signal })
+      throwIfAborted(signal)
+      try {
+        return {
+          PartNumber: partNumber,
+          ...await this.#uploadPartBytes(signature, body, signal),
+        }
+      } catch (err) {
+        await this.#handleError(err)
```

Comment:
nit
```suggestion
        await this.#shouldRetry(err)
```

Reply:
`handleError` implies logging it or something. While we are doing all the rety logic there, so the nit is about making it intention revealing that we retry there